scanner/utils/hash.go

```diff
@@ -108,24 +109,34 @@ func dirFiles(dir string) ([]string, error) {
 
 // generateHash creates hashes for all files along with filenames and generates a hash for the hashes and filenames.
 func generateHash(files []string, open func(string) (io.ReadCloser, error)) (string, error) {
-	h := sha256.New()
 	files = append([]string(nil), files...)
 	sort.Strings(files)
-	for _, file := range files {
-		if strings.Contains(file, "\n") {
-			return "", errors.New("filenames with newlines are not supported")
+
+	mapper := iter.Mapper[string, string]{
+		MaxGoroutines: len(files) / 2, //nolint:mnd
+	}
+
+	results := mapper.Map(files, func(f *string) string {
+		if strings.Contains(*f, "\n") {
+			return ""
 		}
-		r, err := open(file)
+		r, err := open(*f)
```

Comment:
Just a note: if there is anything in the passed open function that is not thread-safe, it can cause race conditions or unwanted behaviour.